backend/src/router.rs

```diff
@@ -0,0 +1,321 @@
+        Self: Sized;
+}
+
+impl<S, B> RouterApiExt<S, B, Infallible> for Router<S, B>
+where
+    B: HttpBody + Send + 'static,
+    S: Clone + Send + Sync + 'static,
+{
+    fn api_route<'a, H, T, Version, ApiDocumentation>(
+        self,
+        path: &str,
+        method: Method,
+        handler: H,
+    ) -> Result<Self, RouteError>
+    where
+        H: Handler<T, S, B>,
+        T: 'static,
+        S: Send + Sync + 'static,
+        Version: ApiVersion<'a>,
+        ApiDocumentation: OpenApi,
+        Self: Sized,
+    {
+        check_api::<_, _, _, H, Version, ApiDocumentation>(
+            path,
+            &*TryInto::<MethodWrapper<PathItemType>>::try_into(method.clone())?,
+        )?;
+        Ok(self.route(
+            path,
+            on(
+                *TryInto::<MethodWrapper<MethodFilter>>::try_into(method)?,
+                handler,
+            ),
+        ))
+    }
+}
+
+/// Check if the following route corresponds with `OpenAPI` declaration
+/// Relies on `operation_id` field, must NOT be changed on handler's declaration
+fn check_api<'a, T, S, B, H, Version, ApiDocumentation>(
+    path: &str,
+    method: &PathItemType,
+) -> Result<(), RouteError>
+where
+    H: Handler<T, S, B>,
+    T: 'static,
+    S: Send + Sync + 'static,
+    ApiDocumentation: OpenApi,
+    Version: ApiVersion<'a>,
+{
+    #[cfg(any(not(debug_assertions), not(feature = "swagger")))]
+    return Ok(());
+    let route = [
+        Version::to_path(),
+        &path
+            .split('/')
+            .map(|arg| {
+                arg.starts_with(':')
+                    .then(|| ["{", &arg[1..], "}"].concat())
+                    .unwrap_or_else(|| arg.to_string())
+            })
+            .collect::<Vec<_>>()
+            .join("/"),
+    ]
+    .concat();
+    let operation_id = ApiDocumentation::openapi()
+        .paths
+        .get_path_item(&route)
+        .ok_or(RouteError::NoRoute)
+        .attach_printable(format!("route: {route}"))?
+        .operations
+        .get(method)
+        .ok_or(RouteError::NoMethod)?
+        .operation_id
+        .clone()
+        .ok_or(RouteError::NoOperation)?;
+    let handler_name = &[std::any::type_name::<H>()
+        .rsplit_once(':')
+        .ok_or(RouteError::InvalidHandler)?
+        .1]
+    .concat();
+
+    operation_id
+        .eq(handler_name)
+        .then_some(())
+        .ok_or(RouteError::NoMatch)
+        .attach_printable(format!("left: {operation_id}, right: {handler_name}"))
+}
+
+// TODO: Restrict input types by some trait?
+pub struct MethodWrapper<T>(T);
+
+impl Deref for MethodWrapper<PathItemType> {
+    type Target = PathItemType;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl Deref for MethodWrapper<MethodFilter> {
+    type Target = MethodFilter;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+// FIXME: Is there a better way to do this?..
+impl TryFrom<Method> for MethodWrapper<PathItemType> {
```

Comment:
Why not `impl TryFrom<Method> for PathItemType`? Why `MethodWrapper<>` was added?

Reply:
Rust blocks implementing foreign traits (aka traits you don't define) for foreign structs. And thus, the MethodWraper was introduced

Reply:
Ok, then it is better to declare just a regular function without any wrapper (`fn convert_method_to_path_item_type(value: Method) -> PathItemType`)

Reply:
Wrapper removed, said functions added